app/models/concerns/facets/managed_host_extensions.rb

```diff
@@ -0,0 +1,66 @@
+require 'facets'
+
+module Facets
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      Facets::ManagedHostExtensions.refresh_facet_relations(self)
+
+      def attributes
+        hash = super
+
+        # include all facet attributes by default
+        facets_with_definitions.each do |facet, facet_definition|
+          hash["#{facet_definition.name}_attributes"] = facet.attributes.reject { |key, _| %w(created_at updated_at).include? key }
+        end
+        hash
+      end
+    end
+
+    def self.refresh_facet_relations(klass)
+      Facets.registered_facets.values.each do |facet_config|
+        self.register_facet_relation(klass, facet_config)
```

Comment:
Both `#register_facet_relation` and `#refresh_facet_relations` are tested here: https://github.com/theforeman/foreman/pull/3076/files#diff-73245f1db9232d228a0ba8a982109379R26

They both are implementation details. Every time a facet registers, the registration work is done by `#register_facet_relation` method. Every time HostManagedExtensions class is included in Host, `#refresh` method will be called.